statsmodels/regression/recursive_ls.py

```diff
@@ -219,7 +252,10 @@ def __init__(self, model, params, filter_results, cov_type='opg',
         super(RecursiveLSResults, self).__init__(
             model, params, filter_results, cov_type, **kwargs)
 
-        self.df_resid = np.inf  # attribute required for wald tests
+        # Since we are overriding params with things that aren't MLE params,
+        # need to adjust df's
+        self.df_model = self.k_diffuse_states
```

Comment:
if you know that `self.k_diffuse_states` does not take constraints into account (or as long as it doesn't), then you could explicitly correct it for the number of constraints, i.e. subtract k_constraints if constraints are not None.

Reply:
Assuming the effect of the (diffuse) initialization goes away as more observations are used, I would go for the later outcome, i.e. use the constraints corrected df_model and df_resid.
I don't really understand how diffuse initialization works, but maybe a comment in the Notes section of the docstring mentioning this would be enough.